app/src/main/java/ch/epfl/sdp/peakar/user/challenge/goal/RemotePointsChallenge.java

```diff
@@ -0,0 +1,92 @@
+import ch.epfl.sdp.peakar.database.DatabaseReference;
+import ch.epfl.sdp.peakar.database.DatabaseSnapshot;
+import ch.epfl.sdp.peakar.user.challenge.Challenge;
+import ch.epfl.sdp.peakar.user.challenge.ChallengeOutcome;
+import ch.epfl.sdp.peakar.user.services.AuthService;
+
+/**
+ * Concrete implementation of <code>PointsChallenge</code> using Database interaction.
+ */
+public class RemotePointsChallenge extends PointsChallenge {
+
+    /**
+     * Constructor for a new concrete points challenge. This constructor needs to be called to retrieve an existing challenge, not to create a new one.
+     * @param id unique identifier of the challenge.
+     * @param users users who joined the challenge.
+     * @param awardPoints points that will be awarded to the winner.
+     * @param goalPoints score to be reached to win.
+     */
+    public RemotePointsChallenge(String id, List<String> users, long awardPoints, long goalPoints) {
+        super(id, users, awardPoints, goalPoints);
+    }
+
+    /**
+     * Generate a new PointsChallenge.
+     * @param founderID ID of the founder of the challenge.
+     * @param goalPoints score to be reached to win.
+     */
+    public static Challenge generateNewChallenge(String founderID, long goalPoints) {
+        // Create a new challenge remotely
+        DatabaseReference challengeRef = Database.getInstance().getReference().child(Database.CHILD_CHALLENGES).push();
+        // Get ID of new challenge
+        String id = challengeRef.getKey();
+        assert id != null;
+        // Add users
+        List<String> users = new ArrayList<>(Collections.singleton(founderID));
+        challengeRef.child(Database.CHILD_USERS).child(founderID).setValue(FOUNDER);
+        // Add prize
+        challengeRef.child(Database.CHILD_CHALLENGE_GOAL).setValue(goalPoints);
+        // Join the new challenge remotely
+        Database.getInstance().getReference().child(Database.CHILD_USERS).child(founderID).child(Database.CHILD_CHALLENGES).child(id).setValue(Database.VALUE_POINTS_CHALLENGE);
+        RemotePointsChallenge newChallenge = new RemotePointsChallenge(id, users, 0L, goalPoints);
+        // Add locally if the founder is the authenticated user
+        if(founderID.equals(AuthService.getInstance().getID())) AuthService.getInstance().getAuthAccount().getChallenges().add(newChallenge);
+        return newChallenge;
+    }
+
+    @Override
+    public ChallengeOutcome join() {
+        // If the authenticated user has already joined this challenge.
+        if(getUsers().contains(AuthService.getInstance().getID())) return ChallengeOutcome.NOT_POSSIBLE;
+        // Join remotely
+        Database.getInstance().getReference().child(Database.CHILD_CHALLENGES).child(getID()).child(Database.CHILD_USERS).child(AuthService.getInstance().getID()).setValue(JOINED);
+        // Add locally
+        AuthService.getInstance().getAuthAccount().getChallenges().add(this);
+        return super.join();
```

Comment:
I think you may be right, but @bgabriel998 is working on that class at the moment, I'm not sure if we should make this change in this PR as it is about the database (all these lines appear as new just because I renamed and moved Remote classes).